woodwork/logical_types.py

```diff
@@ -306,6 +310,16 @@ class EmailAddress(LogicalType):
 
     primary_dtype = "string"
 
+    def validate(self, series, return_indices=False):
+        regex = r"\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b"
+        valid = series.str.fullmatch(regex).astype("boolean")
+
+        if return_indices:
+            return valid.rename(series.name)
+
+        elif (~valid).any():
+            raise ValueError("email address not understood")
```

Comment:
I think this could be a good time to create a custom exception. If users want to catch the validation error, having a try/except on a `ValueError` could end up falling into the except if some part of validation raises an unrelated exception.

Reply:
Will we want custom exceptions for every logical type?

Reply:
Maybe not an exception that's specific to each logical type, but one that indicates the data is invalid for its specified logical type?

Reply:
I see, so maybe we can use `TypeValidationError` for validation errors across logical types?

Reply:
that sounds good to me!